From 080a0662e235bb359ef06f6ba922f56a6040078f Mon Sep 17 00:00:00 2001
From: "Hongzhi.Song" <hongzhi.song@windriver.com>
Date: Tue, 11 Dec 2018 04:54:59 -0500
Subject: [PATCH] tool/kconf_check: modify grep pattern

The cmd line, <grep '^[  ]*\(menu\)*config '>, can't
match all expect config options.

This is because that it is not always space after 'config'
in kernel-source/*/.../Kconfig.
e.g. "config  IP_VS_IPV6" in net/netfilter/ipvs/Kconfig

So we should change the cmd to grep '^[  ]*\(menu\)*config\s'.

Upstream-Status: Submitted
https://lists.yoctoproject.org/pipermail/yocto/2018-December/043564.html

Signed-off-by: Hongzhi.Song <hongzhi.song@windriver.com>
---
 tools/kconf_check | 2 +-
 1 file changed, 1 insertion(+), 1 deletion(-)

diff --git a/tools/kconf_check b/tools/kconf_check
index aa86180..20b49cd 100755
--- a/tools/kconf_check
+++ b/tools/kconf_check
@@ -241,7 +241,7 @@ find ${kernel_source} \
 # known list of all Kconfig* files.  Again, must filter dups.
 rm -f ${LOGDIR}/all.cfg
 for i in `cat ${LOGDIR}/all.kcf` ; do
-    cat ${kernel_source}/$i | grep '^[ 	]*\(menu\)*config ' | \
+    cat ${kernel_source}/$i | grep '^[ 	]*\(menu\)*config\s' | \
 	awk '{print "CONFIG_"$2}' >> ${LOGDIR}/all.cfg
 done
 mv -f ${LOGDIR}/all.cfg ${LOGDIR}/all.cfg~
-- 
2.8.1

